samtranslator/model/api/api_generator.py

```diff
@@ -565,6 +575,18 @@ def _construct_api_domain(  # noqa: too-many-branches
             logical_id = "RecordSetGroup" + logical_id_suffix
 
             record_set_group = route53_record_set_groups.get(logical_id)
+
+            if route53.get("SeparateRecordSets") and not is_intrinsic(route53.get("SeparateRecordSets")):
```

Comment:
Isn't the

```
and not is_intrinsic(route53.get("SeparateRecordSets"))
```

unnecessary? Since it checks `to_be_a_bool`.

Reply:
good point. Removed.